src/Microsoft.AspNetCore.ResponseCaching/ResponseCachingContext.cs

```diff
@@ -227,9 +249,25 @@ internal void FinalizeCachingBody()
         {
             if (State.ShouldCacheResponse && ResponseCacheStream.BufferingEnabled)
             {
-                State.CachedResponse.Body = ResponseCacheStream.BufferedStream.ToArray();
+                if (ResponseCacheStream.BufferedStream.Length >= _options.MinimumSplitBodySize)
```

Comment:
This doesn't give you precise control of your cache entry size because it does not account for the size of the rest of the data (e.g. Headers).